### PR TITLE
[Release] sdk/resourcemanager/authorization/armauthorization/2.0.0

### PR DESCRIPTION
**Pull request by Alancere**
This release help to fix two issues:
- https://github.com/Azure/azure-sdk-for-go/issues/18811
- https://github.com/Azure/azure-sdk-for-go/issues/18674

### CODE REVIEW

**Comment by tadelesh** [comment]
Wait for .Net authorization GA.

**Review by RickWinter — CHANGES_REQUESTED** [review]
v2 releases are painful for customers.  We want to keep this on the v1.x path.

**Comment by tadelesh** [comment]
> v2 releases are painful for customers. We want to keep this on the v1.x path.

The breakings contain required param removal, property ref change and operation removal from swagger. I don't think it is possible to make it 1.x.

**Comment by mblaschke-daimlertruck** [comment]
Any progress here? Is there a way to get this merged in the next days?

**Comment by tadelesh** [comment]
@RickWinter, I've use directive to mitigate most of the breakings. Only one orphan model not used is left. I think it could be ignored. You could see the new generated [SDK](https://github.com/azure-sdk/azure-sdk-for-go/pull/2716) here and directive [here](https://github.com/Azure/azure-rest-api-specs/blob/2c5470e59e29aea941cd6448a5cb6c182e0f834a/specification/authorization/resource-manager/readme.go.md?plain=1#L18-L49). @JeffreyRichter Could you help to confirm whether we could release it with 1.1.0?

**Comment by JeffreyRichter** [comment]
Joel & I spoke about this today. Here's where we're at...
It is a reality today that swaggers have bugs, we generate a client and then the swagger gets fixed causing a breaking change in the client (even though the service itself has not changed).

So, if we ship a client V1.0.0 and then the swagger gets fixed, we ship a new client V1.0.1.
Now, let's say an app uses V1.0.1 and also uses another package that depends on V1.0.0 - the app can't build now because V1.0.0 & V1.0.1 can't be compiled together into a single app - this is a very bad place for a customer to be and so we can't do this and put them there.

So, **any** client library break **must** be done in a new major version so that the new major and old major can be compiled together into a single app.

Obviously, the best thing for us (service team and SDK teams) to do is to better validate our swaggers and fix bugs **before** we create and ship client libraries.

**Comment by mblaschke** [comment]
Any progress here?

**Comment by tadelesh** [comment]
> Any progress here?

We are double confirming with service team and will release the new version soon.

**Comment by tadelesh** [comment]
@RickWinter Please sign off. .NET has released `package-2022-04-01` as stable and we also got confirmation from service team to release Go SDK.

**Review by tadelesh — APPROVED** [review]

**Comment by tadelesh** [comment]
Release request: https://github.com/Azure/sdk-release-request/issues/3207

File: sdk/resourcemanager/authorization/armauthorization/autorest.md [hunk 1]
```diff
@@ -5,8 +5,9 @@
...
 azure-arm: true
 require:
-- https://github.com/Azure/azure-rest-api-specs/blob/08894fa8d66cb44dc62a73f7a09530f905985fa3/specification/authorization/resource-manager/readme.md
-- https://github.com/Azure/azure-rest-api-specs/blob/08894fa8d66cb44dc62a73f7a09530f905985fa3/specification/authorization/resource-manager/readme.go.md
+- https://github.com/Azure/azure-rest-api-specs/blob/53b1affe357b3bfbb53721d0a2002382a046d3b0/specification/authorization/resource-manager/readme.md
+- https://github.com/Azure/azure-rest-api-specs/blob/53b1affe357b3bfbb53721d0a2002382a046d3b0/specification/authorization/resource-manager/readme.go.md
 license-header: MICROSOFT_MIT_NO_VERSION
-module-version: 2.0.0-beta.1
+module-version: 2.0.0
+tag: package-2022-04-01
```
Comment (jhendrixMSFT) [inline sdk/resourcemanager/authorization/armauthorization/autorest.md:12]: What tag was previously used?

Comment (tadelesh) [reply sdk/resourcemanager/authorization/armauthorization/autorest.md:12]: `1.0.0` uses tag `package-2020-10-01`.

**Comment by tadelesh** [comment]
All the breaking changes is from this [PR](https://github.com/Azure/azure-rest-api-specs/pull/19937) and approved in this [PR](https://github.com/Azure/azure-rest-api-specs-pr/pull/6248).
1. `Function *RoleAssignmentsClient.NewListForResourcePager parameter(s) have been changed from (string, string, string, string, string, *RoleAssignmentsClientListForResourceOptions) to (string, string, string, string, *RoleAssignmentsClientListForResourceOptions)`
New swagger [here](https://github.com/Azure/azure-rest-api-specs/blob/1d9c9c68844d2b5883621069d5bf0c958841782f/specification/authorization/resource-manager/Microsoft.Authorization/stable/2022-04-01/authorization-RoleAssignmentsCalls.json#L132). This operation's path params have been changed. Old version [here](https://github.com/Azure/azure-rest-api-specs/blob/1d9c9c68844d2b5883621069d5bf0c958841782f/specification/authorization/resource-manager/Microsoft.Authorization/stable/2015-07-01/authorization-RoleAssignmentsCalls.json#L37).
2. `Type of RoleAssignment.Properties has been changed from *RoleAssignmentPropertiesWithScope to *RoleAssignmentProperties`
`Struct RoleAssignmentPropertiesWithScope has been removed`
`Properties` of `RoleAssignment` model has been changed. Old one [here](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/authorization/resource-manager/Microsoft.Authorization/stable/2015-07-01/authorization-RoleAssignmentsCalls.json#L594). New one [here](https://github.com/Azure/azure-rest-api-specs/blob/1d9c9c68844d2b5883621069d5bf0c958841782f/specification/authorization/resource-manager/Microsoft.Authorization/stable/2022-04-01/authorization-RoleAssignmentsCalls.json#L654).
3. `Function *RoleAssignmentsClient.NewListPager has been removed`
`Struct RoleAssignmentsClientListOptions has been removed`
`Struct RoleAssignmentsClientListResponse has been removed`
An operation has been renamed. Old swagger [here](https://github.com/Azure/azure-rest-api-specs/blob/1d9c9c68844d2b5883621069d5bf0c958841782f/specification/authorization/resource-manager/Microsoft.Authorization/stable/2015-07-01/authorization-RoleAssignmentsCalls.json#L450). New one [here](https://github.com/Azure/azure-rest-api-specs/blob/1d9c9c68844d2b5883621069d5bf0c958841782f/specification/authorization/resource-manager/Microsoft.Authorization/stable/2022-04-01/authorization-RoleAssignmentsCalls.json#L37).
4. Struct PermissionAutoGenerated has been removed
`Permission` model has been changed to align with other `Permission` definition. Old one [here](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/authorization/resource-manager/Microsoft.Authorization/stable/2015-07-01/authorization-RoleDefinitionsCalls.json#L498). New one [here](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/authorization/resource-manager/Microsoft.Authorization/stable/2022-04-01/authorization-RoleDefinitionsCalls.json#L505).

**Comment by tadelesh** [comment]
@jhendrixMSFT @RickWinter @JeffreyRichter I list all the breakings and please help to review.

**Review by jhendrixMSFT — APPROVED** [review]

**Review by JeffreyRichter — APPROVED** [review]

**Pull request merged**